fix: KubectlHandler - insecure kubeconfig warning

mrsiejas opened the pull request:
KubectlHandler started to return insecure kubeconfig file warning starting Kubernetes 1.20
```
2:08:24 PM | CREATE_FAILED | Custom::AWSCDK-EKS-HelmChart | NginxIngressController/Resource/Default
Received response status [FAILED] from custom resource. Message returned: Error: b'WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /tmp/kubeconfig\nWARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /tmp/kubeconfig\nError: UPGRADE FAILED: an
other operation (install/upgrade/rollback) is in progress\n'
```

Fix changes permissions of the file to read and write for the User and removes permissions for Group and Others. 


Fixes #14560

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: d75d065a2ccfb9977b51a3776f622cf1dbe8754a
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f2cefb35-7f62-480e-b0bf-92611de3565d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 50dde3e08e9e345bee45129c8c0bc7bb8f12b86c
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1f9ad3f0-0360-44b3-b81c-66f0b55d9e72%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 6271847cab658faba8576efb95512f0cf240bfd9
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=edae7f52-2ba8-4ed9-9405-834ffbe7b79e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: f11360be2383a371ed892d061c029e734351e0f8
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=73a7edab-f625-4008-8cc4-a4597279d381%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

otaviomacedo reviewed (commented):
Looks good to me, but the tests are failing. Can you please take a look?

mrsiejas commented:
Not sure why, looks like error is being thrown by the legacy handler
```
@aws-cdk/aws-eks-legacy: Resources
@aws-cdk/aws-eks-legacy: [~] AWS::Lambda::Function myClusterResourceHandler19D131C9 
@aws-cdk/aws-eks-legacy:  â””â”€ [~] Code
@aws-cdk/aws-eks-legacy:      â”œâ”€ [~] .S3Bucket:
@aws-cdk/aws-eks-legacy:      â”‚   â””â”€ [~] .Ref:
@aws-cdk/aws-eks-legacy:      â”‚       â”œâ”€ [-] AssetParametersea4957b16062595851e7d293ee45835db05c5693669a729cc02944b6ad19a204S3Bucket371D99F8
@aws-cdk/aws-eks-legacy:      â”‚       â””â”€ [+] AssetParameters1eec89d987dc6420031e92f6d15199c2d8f34a844ef745b60c09b0afd32f5070S3BucketA540BC94
@aws-cdk/aws-eks-legacy:      â””â”€ [~] .S3Key:
@aws-cdk/aws-eks-legacy:          â””â”€ [~] .Fn::Join:
@aws-cdk/aws-eks-legacy:              â””â”€ @@ -8,7 +8,7 @@
@aws-cdk/aws-eks-legacy:                 [ ]   "Fn::Split": [
@aws-cdk/aws-eks-legacy:                 [ ]     "||",
@aws-cdk/aws-eks-legacy:                 [ ]     {
@aws-cdk/aws-eks-legacy:                 [-]       "Ref": "AssetParametersea4957b16062595851e7d293ee45835db05c5693669a729cc02944b6ad19a204S3VersionKeyFDCB25DD"
@aws-cdk/aws-eks-legacy:                 [+]       "Ref": "AssetParameters1eec89d987dc6420031e92f6d15199c2d8f34a844ef745b60c09b0afd32f5070S3VersionKey2FE96A49"
@aws-cdk/aws-eks-legacy:                 [ ]     }
@aws-cdk/aws-eks-legacy:                 [ ]   ]
@aws-cdk/aws-eks-legacy:                 [ ] }
@aws-cdk/aws-eks-legacy:                 @@ -21,7 +21,7 @@
@aws-cdk/aws-eks-legacy:                 [ ]   "Fn::Split": [
@aws-cdk/aws-eks-legacy:                 [ ]     "||",
@aws-cdk/aws-eks-legacy:                 [ ]     {
@aws-cdk/aws-eks-legacy:                 [-]       "Ref": "AssetParametersea4957b16062595851e7d293ee45835db05c5693669a729cc02944b6ad19a204S3VersionKeyFDCB25DD"
@aws-cdk/aws-eks-legacy:                 [+]       "Ref": "AssetParameters1eec89d987dc6420031e92f6d15199c2d8f34a844ef745b60c09b0afd32f5070S3VersionKey2FE96A49"
@aws-cdk/aws-eks-legacy:                 [ ]     }
@aws-cdk/aws-eks-legacy:                 [ ]   ]
@aws-cdk/aws-eks-legacy:                 [ ] }
@aws-cdk/aws-eks-legacy: [~] AWS::Lambda::Function myClusterKubernetesResourceHandler50297E32 
@aws-cdk/aws-eks-legacy:  â””â”€ [~] Code
@aws-cdk/aws-eks-legacy:      â”œâ”€ [~] .S3Bucket:
@aws-cdk/aws-eks-legacy:      â”‚   â””â”€ [~] .Ref:
@aws-cdk/aws-eks-legacy:      â”‚       â”œâ”€ [-] AssetParameters640847533c8a00b3133aeb128edcac41fb7b60349c9e18764fcf7ea4af14d444S3Bucket919126CB
@aws-cdk/aws-eks-legacy:      â”‚       â””â”€ [+] AssetParametersd40ac2278fbe4accc92f2ef7677a13a067d9ed479ec3e84d89a49042f89c6dc8S3Bucket8D6F7A44
@aws-cdk/aws-eks-legacy:      â””â”€ [~] .S3Key:
@aws-cdk/aws-eks-legacy:          â””â”€ [~] .Fn::Join:
@aws-cdk/aws-eks-legacy:              â””â”€ @@ -8,7 +8,7 @@
@aws-cdk/aws-eks-legacy:                 [ ]   "Fn::Split": [
@aws-cdk/aws-eks-legacy:                 [ ]     "||",
@aws-cdk/aws-eks-legacy:                 [ ]     {
@aws-cdk/aws-eks-legacy:                 [-]       "Ref": "AssetParameters640847533c8a00b3133aeb128edcac41fb7b60349c9e18764fcf7ea4af14d444S3VersionKey529BEF54"
@aws-cdk/aws-eks-legacy:                 [+]       "Ref": "AssetParametersd40ac2278fbe4accc92f2ef7677a13a067d9ed479ec3e84d89a49042f89c6dc8S3VersionKeyAA64BD77"
@aws-cdk/aws-eks-legacy:                 [ ]     }
@aws-cdk/aws-eks-legacy:                 [ ]   ]
@aws-cdk/aws-eks-legacy:                 [ ] }
@aws-cdk/aws-eks-legacy:                 @@ -21,7 +21,7 @@
@aws-cdk/aws-eks-legacy:                 [ ]   "Fn::Split": [
@aws-cdk/aws-eks-legacy:                 [ ]     "||",
@aws-cdk/aws-eks-legacy:                 [ ]     {
@aws-cdk/aws-eks-legacy:                 [-]       "Ref": "AssetParameters640847533c8a00b3133aeb128edcac41fb7b60349c9e18764fcf7ea4af14d444S3VersionKey529BEF54"
@aws-cdk/aws-eks-legacy:                 [+]       "Ref": "AssetParametersd40ac2278fbe4accc92f2ef7677a13a067d9ed479ec3e84d89a49042f89c6dc8S3VersionKeyAA64BD77"
@aws-cdk/aws-eks-legacy:                 [ ]     }
@aws-cdk/aws-eks-legacy:                 [ ]   ]
@aws-cdk/aws-eks-legacy:                 [ ] }
@aws-cdk/aws-eks-legacy: Error: Some stacks have changed. To verify that they still deploy successfully, run: 'yarn integ integ.eks-cluster.defaults.js integ.eks-cluster.lit.js integ.eks-helm.lit.js integ.eks-kubectl.lit.js integ.eks-spot.js'
@aws-cdk/aws-eks-legacy:     at main (/codebuild/output/src704439715/src/github.com/aws/aws-cdk/tools/cdk-integ-tools/bin/cdk-integ-assert.js:36:15)
@aws-cdk/aws-eks-legacy: Error: cdk-integ-assert exited with error code 1
@aws-cdk/aws-eks-legacy: Tests failed. Total time (9.0s) | /codebuild/output/src704439715/src/github.com/aws/aws-cdk/node_modules/nyc/bin/nyc.js (5.1s) | cdk-integ-assert (3.9s)
@aws-cdk/aws-eks-legacy: !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
@aws-cdk/aws-eks-legacy: error Command failed with exit code 1.
@aws-cdk/aws-eks-legacy: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@aws-cdk/aws-eks-legacy: error Command failed with exit code 1.
@aws-cdk/aws-eks-legacy: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@aws-cdk/aws-eks-legacy: error Command failed with exit code 1.
@aws-cdk/aws-eks-legacy: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run build+test+extract exited 1 in '@aws-cdk/aws-eks-legacy'
lerna WARN complete Waiting for 7 child processes to exit. CTRL-C to exit immediately.

real	14m45.261s
user	281m29.814s
sys	21m52.770s
âŒ  Last command failed. Scroll up to see errors in log (search for '!!!!!!!!').

[Container] 2021/08/16 08:26:34 Command did not exit successfully /bin/bash ./build.sh --extract && git diff-index --exit-code --ignore-space-at-eol --stat HEAD exit status 1
[Container] 2021/08/16 08:26:34 Phase complete: BUILD State: FAILED
```
Not sure what it does without reading all code. Will try removing changes from file within aws-eks-legacy folder.

otaviomacedo commented:
This should be enough for you to solve it:

> @aws-cdk/aws-eks-legacy: Error: Some stacks have changed. To verify that they still deploy successfully, run: 'yarn integ integ.eks-cluster.defaults.js integ.eks-cluster.lit.js integ.eks-helm.lit.js integ.eks-kubectl.lit.js integ.eks-spot.js'

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 10ea086d2bf0dab6aecff5f133f443b8dce991cb
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1df631a7-8df8-42b0-8c42-12d73dee9973%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: f02ded461b31d481350d9b0bd0b38aafb7e8a51d
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cfca0fb4-2bc5-4f3c-b883-5a3d6e8fcc2a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 746c00a5aa766a5e8ef06aa431c19d0573930fb5
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0bbeebb6-8d40-4de4-a21f-c0a550373eba%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
